# **简介**

## **1980 年至 2023 年的热门电子游戏数据**


简介：数据集包含从 1980 年到 2023 年的视频游戏列表，并提供发布日期、用户评论评级和评论家评论评级等信息。

变量含义：
- Title：游戏标题
- Release Date：游戏首个版本发布日期
- Team：游戏开发团队
- Rating：平均评分
- Times Listed：列出此游戏的用户数量
- Number of Reviews：用户提供的评论数量
- Genres：游戏所属的所有类型/流派
- Summary：团队提供的摘要/概述
- Reviews：用户的评价/评论

## **读取数据**

导入pandas库，使用Pandas的`read_csv`函数，将原始数据文件“games.csv”里的数据内容，解析为DatatFrame，并赋值给变量`original_data`。

In [1]:
import pandas as pd

In [2]:
original_data = pd.read_csv("../[赠送] 练习数据集（持续更新）/games.csv")
original_data

Unnamed: 0                                    Title  Release Date  \
0              0                               Elden Ring  Feb 25, 2022   
1              1                                    Hades  Dec 10, 2019   
2              2  The Legend of Zelda: Breath of the Wild  Mar 03, 2017   
3              3                                Undertale  Sep 15, 2015   
4              4                            Hollow Knight  Feb 24, 2017   
...          ...                                      ...           ...   
1507        1507             Back to the Future: The Game  Dec 22, 2010   
1508        1508                        Team Sonic Racing  May 21, 2019   
1509        1509                           Dragon's Dogma  May 22, 2012   
1510        1510                          Baldur's Gate 3  Oct 06, 2020   
1511        1511                 The LEGO Movie Videogame  Feb 04, 2014   

                                                   Team  Rating Times Listed  \
0        ['Bandai Namco Entertainment', 'FromSoftware']     4.5         3.9K   
1                                  ['Supergiant Games']     4.3         2.9K   
2     ['Nintendo', 'Nintendo EPD Production Group No...     4.4         4.3K   
3                                    ['tobyfox', '8-4']     4.2         3.5K   
4                                       ['Team Cherry']     4.4           3K   
...                                                 ...     ...          ...   
1507                                 ['Telltale Games']     3.2           94   
1508                           ['Sumo Digital', 'Sega']     2.9          264   
1509                                         ['Capcom']     3.7          210   
1510                                 ['Larian Studios']     4.1          165   
1511                          ['WB Games', 'TT Fusion']     2.8          184   

     Number of Reviews                                             Genres  \
0                 3.9K                               ['Adventure', 'RPG']   
1                 2.9K           ['Adventure', 'Brawler', 'Indie', 'RPG']   
2                 4.3K                               ['Adventure', 'RPG']   
3                 3.5K  ['Adventure', 'Indie', 'RPG', 'Turn Based Stra...   
4                   3K                 ['Adventure', 'Indie', 'Platform']   
...                ...                                                ...   
1507                94                   ['Adventure', 'Point-and-Click']   
1508               264                               ['Arcade', 'Racing']   
1509               210                                 ['Brawler', 'RPG']   
1510               165  ['Adventure', 'RPG', 'Strategy', 'Tactical', '...   
1511               184                            ['Adventure', 'Puzzle']   

                                                Summary  \
0     Elden Ring is a fantasy, action and open world...   
1     A rogue-lite hack and slash dungeon crawler in...   
2     The Legend of Zelda: Breath of the Wild is the...   
3     A small child falls into the Underground, wher...   
4     A 2D metroidvania with an emphasis on close co...   
...                                                 ...   
1507  Back to the Future: The Game is one of Telltal...   
1508  Team Sonic Racing combines the best elements o...   
1509  Set in a huge open world, Dragon’s Dogma: Dark...   
1510  An ancient evil has returned to Baldur's Gate,...   
1511  Join Emmet and an unlikely group of resistance...   

                                                Reviews Plays Playing  \
0     ["The first playthrough of elden ring is one o...   17K    3.8K   
1     ['convinced this is a roguelike for people who...   21K    3.2K   
2     ['This game is the game (that is not CS:GO) th...   30K    2.5K   
3     ['soundtrack is tied for #1 with nier automata...   28K     679   
4     ["this games worldbuilding is incredible, with...   21K    2.4K   
...                                                 ...   ...     ...   

## **评估数据**

这一部分，我将对在上一步分建立的`original_data`这个DataFrame所包含的数据进行评估。

评估主要从两个方面进行：结构和内容，即整齐度和干净度。数据的结构性问题指不符合“每列是一个变量，每行是一个观察值，每个单元格是一个值”
这三个标准，数据的内容性问题包括存在丢失的数据、重复数据、无效数据等等。

In [3]:
pd.set_option("display.max_colwidth",20)
original_data.sample(10)

Unnamed: 0                Title  Release Date                 Team  \
1410        1410  The Secret of Mo...  Oct 01, 1990  ['Lucasfilm Game...   
811          811  Sekiro: Shadows ...  Mar 22, 2019  ['FromSoftware',...   
366          366  Resident Evil Vi...  May 06, 2021           ['Capcom']   
880          880                 Sifu  Feb 08, 2022  ['Sloclap', 'Kep...   
72            72           EarthBound  Aug 27, 1994  ['Nintendo', 'Cr...   
554          554  Middle-earth: Sh...  Sep 30, 2014  ['Monolith Produ...   
1407        1407  JoJo's Bizarre A...  Dec 17, 2015  ['CyberConnect2'...   
772          772  The Legend of He...  Sep 25, 2014  ['NIS America', ...   
783          783        Metroid Dread  Oct 07, 2021  ['Nintendo', 'Me...   
1010        1010  Marvel's Spider-...  Nov 12, 2020  ['Insomniac Game...   

      Rating Times Listed Number of Reviews               Genres  \
1410     4.1          312               312  ['Adventure', 'P...   
811      4.4         2.3K              2.3K  ['Adventure', 'B...   
366      3.9         1.8K              1.8K  ['Adventure', 'S...   
880      3.8          633               633  ['Brawler', 'Ind...   
72       4.2         1.9K              1.9K  ['Adventure', 'R...   
554      3.4          524               524  ['Adventure', 'B...   
1407     2.8          106               106  ['Arcade', 'Figh...   
772      3.7          401               401  ['Adventure', 'R...   
783      4.3         2.1K              2.1K  ['Adventure', 'P...   
1010     4.2          335               335  ['Adventure', 'B...   

                  Summary              Reviews Plays Playing Backlogs Wishlist  
1410  I cursed my luck...  ["I actually pla...  1.4K      13      418      298  
811   Enter a dark and...  ["Im waiting for...   14K     919     4.8K     3.4K  
366   Resident Evil Vi...  ['This game is w...  9.9K     333     2.8K     2.8K  
880   A third person a...  ["When I first p...  2.7K     283     1.7K     2.3K  
72    A turn-based JRP...  ['i think the ro...    9K     600     4.8K     1.8K  
554   Middle-earth: Sh...  ['It has combat ...  8.4K     106     1.7K      478  
1407  JoJo’s Bizarre A...  ['Meh asf', 'I m...  1.7K      36      276      196  
772   The Legend of He...  ['This game is s...  1.7K      45      707      319  
783   Join intergalact...  ['Have only been...  9.2K     759     3.4K     3.3K  
1010  Included in the ...  ['- 32.3 Hours t...  2.8K     206      574      472

从抽样的10行数据来看，数据符合“每列是一个变量，每行是一个观察值，每个单元格是一个值”，具体来看每行是关于某游戏的记录，每列是游戏各个属性的变量，因此不存在结构性问题

### **评估数据干净度**

In [4]:
original_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1512 entries, 0 to 1511
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         1512 non-null   int64  
 1   Title              1512 non-null   object 
 2   Release Date       1512 non-null   object 
 3   Team               1511 non-null   object 
 4   Rating             1499 non-null   float64
 5   Times Listed       1512 non-null   object 
 6   Number of Reviews  1512 non-null   object 
 7   Genres             1512 non-null   object 
 8   Summary            1511 non-null   object 
 9   Reviews            1512 non-null   object 
 10  Plays              1512 non-null   object 
 11  Playing            1512 non-null   object 
 12  Backlogs           1512 non-null   object 
 13  Wishlist           1512 non-null   object 
dtypes: float64(1), int64(1), object(12)
memory usage: 165.5+ KB


从输出结果来看，数据共有1512条观察值，而`Team`、`Rating` 、`Summary`变量存在缺失值。

#### **评估缺失数据**

在了解`Team`存在缺失值后,根据条件提取出缺失观察值:

In [5]:
original_data[original_data["Team"].isnull()]

Unnamed: 0                Title  Release Date Team  Rating Times Listed  \
1245        1245  NEET Girl Date N...  Oct 21, 2022  NaN     2.7           21   

     Number of Reviews            Genres              Summary  \
1245                21  ['Visual Novel']  Your friend sets...   

                  Reviews Plays Playing Backlogs Wishlist  
1245  ['this sucked. "...   106       1       44       42

`Team`表示其游戏制作公司，从输出结果来看，缺失`Team`变量的数据只有一条，并且该变量的缺失对此次数据整理并无影响，所以可以忽略。

`Rating`变量也提取观察，判断其缺失情况:

In [6]:
original_data[original_data["Rating"].isnull()]

Unnamed: 0                Title     Release Date                 Team  \
587          587    Final Fantasy XVI     Jun 22, 2023  ['Square Enix', ...   
649          649    Death Stranding 2  releases on TBD  ['Kojima Product...   
713          713  Final Fantasy VI...     Dec 31, 2023      ['Square Enix']   
719          719            Lies of P     Aug 01, 2023  ['NEOWIZ', 'Roun...   
726          726                Judas     Mar 31, 2025  ['Ghost Story Ga...   
746          746  Like a Dragon Ga...     Dec 31, 2023  ['Ryū Ga Gotoku ...   
972          972  The Legend of Ze...     May 12, 2023  ['Nintendo', 'Ni...   
1130        1130  Star Wars Jedi: ...     Apr 28, 2023  ['Respawn Entert...   
1160        1160  We Love Katamari...     Jun 02, 2023  ['Bandai Namco E...   
1202        1202           Earthblade     Dec 31, 2024  ['Extremely OK G...   
1305        1305            Diablo IV     Jun 06, 2023  ['Blizzard Enter...   
1341        1341     Pac-Man in a Tin     Mar 22, 2023   ['Fizz Creations']   
1349        1349         Pokémon Añil     Mar 23, 2023        ['Eric Lost']   

      Rating Times Listed Number of Reviews               Genres  \
587      NaN          422               422              ['RPG']   
649      NaN          105               105  ['Adventure', 'S...   
713      NaN          192               192                   []   
719      NaN          175               175              ['RPG']   
726      NaN           90                90  ['Adventure', 'S...   
746      NaN          118               118  ['Adventure', 'B...   
972      NaN          581               581  ['Adventure', 'R...   
1130     NaN          250               250        ['Adventure']   
1160     NaN           51                51  ['Adventure', 'P...   
1202     NaN           83                83  ['Adventure', 'I...   
1305     NaN          148               148   ['Brawler', 'RPG']   
1341     NaN            0                 0           ['Arcade']   
1349     NaN            2                 2              ['RPG']   

                  Summary              Reviews Plays Playing Backlogs Wishlist  
587   Final Fantasy XV...                   []    37      10      732     2.4K  
649                   NaN                   []     3       0      209      644  
713   This next standa...                   []    20       3      354     1.1K  
719   Inspired by the ...                   []     5       0      260      939  
726   A disintegrating...                   []     1       0       92      437  
746   This game covers...                   []     2       1      145      588  
972   The Legend of Ze...                   []    72       6     1.6K     5.4K  
1130  The story of Cal...                   []    13       2      367     1.4K  
1160  We Love Katamari...                   []     3       0       74      291  
1202  You are Névoa, a...                   []     0       1      103      529  
1305  Endless demons t...                   []    19      10      179      677  
1341  Pac-Man in a Tin...                   []     1       0        1        2  
1349  Pokémon Añil is ...  ['pokemon HWHAT'...     1       0        1        7

`Rating`变量是针对某款游戏的平均评分；从输出结果来看，`Rating`变量为空的数据有13条，且多数都是游戏发售日期待定或是跳票延期。
因此没有平均分和评价，所以对本次游戏统计信息没有影响，可以将`Rating`变量为空的游戏条目剔除。

`Summary`变量存在缺失值，提取并观察情况：

In [7]:
original_data[original_data["Summary"].isnull()]

Unnamed: 0              Title     Release Date                 Team  \
649         649  Death Stranding 2  releases on TBD  ['Kojima Product...   

     Rating Times Listed Number of Reviews               Genres Summary  \
649     NaN          105               105  ['Adventure', 'S...     NaN   

    Reviews Plays Playing Backlogs Wishlist  
649      []     3       0      209      644

`Summary`变量是对该游戏的概述；从输出结果来看符合变量为空的就1条，且该游戏属于未明确发售日期，因此条该数据在本次统计中没有意义，可以将其剔除。

#### **评估重复数据**

根据数据变量来看，`Title`变量是唯一标识符，表示该游戏的名称；可能存在重复的可能性，因此将对其进行查重。

In [9]:
original_data["Title"].duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
1507    False
1508    False
1509    False
1510    False
1511    False
Name: Title, Length: 1512, dtype: bool

In [10]:
original_data["Team"].duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
1507     True
1508    False
1509     True
1510    False
1511    False
Name: Team, Length: 1512, dtype: bool

In [15]:
original_data[original_data.duplicated(subset=["Team"])]

Unnamed: 0                Title  Release Date                 Team  \
18            18               Portal  Oct 10, 2007  ['Valve', 'Elect...   
25            25  God of War Ragnarök  Nov 09, 2022  ['Sony Interacti...   
30            30       Dark Souls III  Mar 24, 2016  ['Bandai Namco E...   
56            56      Devil May Cry 5  Mar 08, 2019  ['Capcom', 'Capc...   
64            64  The Legend of Ze...  Nov 21, 1998  ['Nintendo', 'Ni...   
...          ...                  ...           ...                  ...   
1502        1502  Billy Hatcher an...  Sep 23, 2003  ['Sonic Team', '...   
1503        1503  Tekken Tag Tourn...  Sep 14, 2011  ['Bandai Namco E...   
1504        1504          Bloons TD 5  Dec 14, 2011       ['Ninja Kiwi']   
1507        1507  Back to the Futu...  Dec 22, 2010   ['Telltale Games']   
1509        1509       Dragon's Dogma  May 22, 2012           ['Capcom']   

      Rating Times Listed Number of Reviews               Genres  \
18       4.1           2K                2K  ['Platform', 'Pu...   
25       4.4         1.7K              1.7K  ['Adventure', 'B...   
30       4.2         2.4K              2.4K  ['Adventure', 'R...   
56       4.3         1.9K              1.9K  ['Adventure', 'B...   
64       4.3         2.2K              2.2K  ['Adventure', 'R...   
...      ...          ...               ...                  ...   
1502     3.1          133               133         ['Platform']   
1503     3.8          211               211         ['Fighting']   
1504     3.6           62                62  ['Simulator', 'S...   
1507     3.2           94                94  ['Adventure', 'P...   
1509     3.7          210               210   ['Brawler', 'RPG']   

                  Summary              Reviews Plays Playing Backlogs Wishlist  
18    Waking up in a s...  ['did somebody s...   28K     244     2.7K     1.1K  
25    God of War: Ragn...  ['Amazang !', 'D...  5.3K     801       2K     3.3K  
30    Dark Souls conti...  ['The only souls...   19K     851     4.3K     2.2K  
56    Devil May Cry 5 ...  ['Goated', 'Comb...   12K     524     4.1K     2.4K  
64    The Legend of Ze...  ["I was a dumb k...   17K     433       3K     1.4K  
...                   ...                  ...   ...     ...      ...      ...  
1502  He may wear a Ro...  ['Truly a very l...   797      12      322      212  
1503  A highly competi...  ['Jun kazama sua...  1.7K       7      140       95  
1504  Five-star tower ...  ['My favorite th...  1.6K      15       46       13  
1507  Back to the Futu...  ['Very enjoyable...   763       5      223       67  
1509  Set in a huge op...  ['Underrated.', ...  1.1K      45      487      206  

[747 rows x 14 columns]

从输出结果可以看到，`Title`变量；游戏名称不存在重复的数据。`Team`变量；开发团队有重复的数据，但同一个公司可以开发多款游戏，因此`Team`可以存在重复。

那么针对该数据集，我们后续不需要进行去重复数据的操作。

#### **评估不一致数据**

不一致数据可能存在于`Genres`变量中，我们需要查看是否是多个不同值指代同一种游戏类型。

In [16]:
original_data["Genres"].value_counts()

Genres
['Adventure', 'RPG']                               154
['Adventure', 'Platform']                          107
['Adventure', 'Shooter']                            82
['Shooter']                                         72
['Adventure', 'RPG', 'Turn Based Strategy']         48
                                                  ... 
['Adventure', 'Brawler', 'Fighting', 'RPG']          1
['Real Time Strategy', 'Simulator', 'Strategy']      1
['Adventure', 'Arcade', 'Platform', 'Shooter']       1
['Fighting', 'Sport']                                1
['Simulator', 'Strategy', 'Tactical']                1
Name: count, Length: 255, dtype: int64

从输出结果来看，一款游戏可以有多种类型和风格，因此我们不会对`Genres`做后续操作。

#### **评估无效或错误数据**

In [18]:
original_data["Rating"].describe()

count    1499.000000
mean        3.719346
std         0.532608
min         0.700000
25%         3.400000
50%         3.800000
75%         4.100000
max         4.800000
Name: Rating, dtype: float64

`Rating`变量表示；游戏评分，从输出结果来看，该数据集中`Rating`变量条目中并无错误数据。

缺失的数据在上文中有过评估，并将在清理数据部分中进行处理。

## **清理数据**

根据前评估部分得到的结论，我们需要进行的数据清洗包括：


-  把`Unnamed:0`变量重名为`Data`
-  把`Data`变量设为新的索引
-  把`Rating`变量缺失的观察值删除

 为了区分开经过清理的数据和原始的数据，我们创建新的变量`cleande_data`,让它为`Original_data`复制出的副本。我们之后的清理步骤都将被运用在`cleande_data`上。

In [19]:
cleande_data = original_data.copy()
cleande_data.head()

Unnamed: 0                Title  Release Date                 Team  Rating  \
0           0           Elden Ring  Feb 25, 2022  ['Bandai Namco E...     4.5   
1           1                Hades  Dec 10, 2019  ['Supergiant Gam...     4.3   
2           2  The Legend of Ze...  Mar 03, 2017  ['Nintendo', 'Ni...     4.4   
3           3            Undertale  Sep 15, 2015   ['tobyfox', '8-4']     4.2   
4           4        Hollow Knight  Feb 24, 2017      ['Team Cherry']     4.4   

  Times Listed Number of Reviews               Genres              Summary  \
0         3.9K              3.9K  ['Adventure', 'R...  Elden Ring is a ...   
1         2.9K              2.9K  ['Adventure', 'B...  A rogue-lite hac...   
2         4.3K              4.3K  ['Adventure', 'R...  The Legend of Ze...   
3         3.5K              3.5K  ['Adventure', 'I...  A small child fa...   
4           3K                3K  ['Adventure', 'I...  A 2D metroidvani...   

               Reviews Plays Playing Backlogs Wishlist  
0  ["The first play...   17K    3.8K     4.6K     4.8K  
1  ['convinced this...   21K    3.2K     6.3K     3.6K  
2  ['This game is t...   30K    2.5K       5K     2.6K  
3  ['soundtrack is ...   28K     679     4.9K     1.8K  
4  ["this games wor...   21K    2.4K     8.3K     2.3K

把`Unnamed:0`变量重名为`Data`：

In [27]:
cleande_data.rename(columns={"Unnamed: 0":"Data"},inplace=True)
cleande_data

Title  Release Date                 Team  Rating  \
Data                                                                   
0              Elden Ring  Feb 25, 2022  ['Bandai Namco E...     4.5   
1                   Hades  Dec 10, 2019  ['Supergiant Gam...     4.3   
2     The Legend of Ze...  Mar 03, 2017  ['Nintendo', 'Ni...     4.4   
3               Undertale  Sep 15, 2015   ['tobyfox', '8-4']     4.2   
4           Hollow Knight  Feb 24, 2017      ['Team Cherry']     4.4   
...                   ...           ...                  ...     ...   
1507  Back to the Futu...  Dec 22, 2010   ['Telltale Games']     3.2   
1508    Team Sonic Racing  May 21, 2019  ['Sumo Digital',...     2.9   
1509       Dragon's Dogma  May 22, 2012           ['Capcom']     3.7   
1510      Baldur's Gate 3  Oct 06, 2020   ['Larian Studios']     4.1   
1511  The LEGO Movie V...  Feb 04, 2014  ['WB Games', 'TT...     2.8   

     Times Listed Number of Reviews               Genres              Summary  \
Data                                                                            
0            3.9K              3.9K  ['Adventure', 'R...  Elden Ring is a ...   
1            2.9K              2.9K  ['Adventure', 'B...  A rogue-lite hac...   
2            4.3K              4.3K  ['Adventure', 'R...  The Legend of Ze...   
3            3.5K              3.5K  ['Adventure', 'I...  A small child fa...   
4              3K                3K  ['Adventure', 'I...  A 2D metroidvani...   
...           ...               ...                  ...                  ...   
1507           94                94  ['Adventure', 'P...  Back to the Futu...   
1508          264               264  ['Arcade', 'Raci...  Team Sonic Racin...   
1509          210               210   ['Brawler', 'RPG']  Set in a huge op...   
1510          165               165  ['Adventure', 'R...  An ancient evil ...   
1511          184               184  ['Adventure', 'P...  Join Emmet and a...   

                  Reviews Plays Playing Backlogs Wishlist  
Data                                                       
0     ["The first play...   17K    3.8K     4.6K     4.8K  
1     ['convinced this...   21K    3.2K     6.3K     3.6K  
2     ['This game is t...   30K    2.5K       5K     2.6K  
3     ['soundtrack is ...   28K     679     4.9K     1.8K  
4     ["this games wor...   21K    2.4K     8.3K     2.3K  
...                   ...   ...     ...      ...      ...  
1507  ['Very enjoyable...   763       5      223       67  
1508  ['jogo morto mas...  1.5K      49      413      107  
1509  ['Underrated.', ...  1.1K      45      487      206  
1510  ['Bu türe bu oyu...   269      79      388      602  
1511  ['Legal', 'Prett...  1.7K      11      239       73  

[1499 rows x 13 columns]

把`Data`变量设为新的索引：

In [21]:
cleande_data.set_index("Data",inplace=True)
cleande_data

Title  Release Date                 Team  Rating  \
Data                                                                   
0              Elden Ring  Feb 25, 2022  ['Bandai Namco E...     4.5   
1                   Hades  Dec 10, 2019  ['Supergiant Gam...     4.3   
2     The Legend of Ze...  Mar 03, 2017  ['Nintendo', 'Ni...     4.4   
3               Undertale  Sep 15, 2015   ['tobyfox', '8-4']     4.2   
4           Hollow Knight  Feb 24, 2017      ['Team Cherry']     4.4   
...                   ...           ...                  ...     ...   
1507  Back to the Futu...  Dec 22, 2010   ['Telltale Games']     3.2   
1508    Team Sonic Racing  May 21, 2019  ['Sumo Digital',...     2.9   
1509       Dragon's Dogma  May 22, 2012           ['Capcom']     3.7   
1510      Baldur's Gate 3  Oct 06, 2020   ['Larian Studios']     4.1   
1511  The LEGO Movie V...  Feb 04, 2014  ['WB Games', 'TT...     2.8   

     Times Listed Number of Reviews               Genres              Summary  \
Data                                                                            
0            3.9K              3.9K  ['Adventure', 'R...  Elden Ring is a ...   
1            2.9K              2.9K  ['Adventure', 'B...  A rogue-lite hac...   
2            4.3K              4.3K  ['Adventure', 'R...  The Legend of Ze...   
3            3.5K              3.5K  ['Adventure', 'I...  A small child fa...   
4              3K                3K  ['Adventure', 'I...  A 2D metroidvani...   
...           ...               ...                  ...                  ...   
1507           94                94  ['Adventure', 'P...  Back to the Futu...   
1508          264               264  ['Arcade', 'Raci...  Team Sonic Racin...   
1509          210               210   ['Brawler', 'RPG']  Set in a huge op...   
1510          165               165  ['Adventure', 'R...  An ancient evil ...   
1511          184               184  ['Adventure', 'P...  Join Emmet and a...   

                  Reviews Plays Playing Backlogs Wishlist  
Data                                                       
0     ["The first play...   17K    3.8K     4.6K     4.8K  
1     ['convinced this...   21K    3.2K     6.3K     3.6K  
2     ['This game is t...   30K    2.5K       5K     2.6K  
3     ['soundtrack is ...   28K     679     4.9K     1.8K  
4     ["this games wor...   21K    2.4K     8.3K     2.3K  
...                   ...   ...     ...      ...      ...  
1507  ['Very enjoyable...   763       5      223       67  
1508  ['jogo morto mas...  1.5K      49      413      107  
1509  ['Underrated.', ...  1.1K      45      487      206  
1510  ['Bu türe bu oyu...   269      79      388      602  
1511  ['Legal', 'Prett...  1.7K      11      239       73  

[1512 rows x 13 columns]

把`Rating`变量缺失的观察值删除,并查看删除后该列空缺值的和:

In [24]:
cleande_data.dropna(subset=["Rating"],inplace=True)

In [25]:
cleande_data["Rating"].isnull().sum()

np.int64(0)

In [26]:
cleande_data[cleande_data["Rating"].isnull()]

Empty DataFrame
Columns: [Title, Release Date, Team, Rating, Times Listed, Number of Reviews, Genres, Summary, Reviews, Plays, Playing, Backlogs, Wishlist]
Index: []

## **保存清理后的数据**

完成数据清理后，把干净整齐的数据保存到新的文件里，文件名为`games_1980s_2023s_data_cleaned.csv`。

In [28]:
cleande_data.head()

Title  Release Date                 Team  Rating  \
Data                                                                   
0              Elden Ring  Feb 25, 2022  ['Bandai Namco E...     4.5   
1                   Hades  Dec 10, 2019  ['Supergiant Gam...     4.3   
2     The Legend of Ze...  Mar 03, 2017  ['Nintendo', 'Ni...     4.4   
3               Undertale  Sep 15, 2015   ['tobyfox', '8-4']     4.2   
4           Hollow Knight  Feb 24, 2017      ['Team Cherry']     4.4   

     Times Listed Number of Reviews               Genres              Summary  \
Data                                                                            
0            3.9K              3.9K  ['Adventure', 'R...  Elden Ring is a ...   
1            2.9K              2.9K  ['Adventure', 'B...  A rogue-lite hac...   
2            4.3K              4.3K  ['Adventure', 'R...  The Legend of Ze...   
3            3.5K              3.5K  ['Adventure', 'I...  A small child fa...   
4              3K                3K  ['Adventure', 'I...  A 2D metroidvani...   

                  Reviews Plays Playing Backlogs Wishlist  
Data                                                       
0     ["The first play...   17K    3.8K     4.6K     4.8K  
1     ['convinced this...   21K    3.2K     6.3K     3.6K  
2     ['This game is t...   30K    2.5K       5K     2.6K  
3     ['soundtrack is ...   28K     679     4.9K     1.8K  
4     ["this games wor...   21K    2.4K     8.3K     2.3K

In [30]:
cleande_data.to_csv("games_1980s_2023s_data_cleaned.csv")

In [31]:
pd.read_csv("games_1980s_2023s_data_cleaned.csv").head()

Data                Title  Release Date                 Team  Rating  \
0     0           Elden Ring  Feb 25, 2022  ['Bandai Namco E...     4.5   
1     1                Hades  Dec 10, 2019  ['Supergiant Gam...     4.3   
2     2  The Legend of Ze...  Mar 03, 2017  ['Nintendo', 'Ni...     4.4   
3     3            Undertale  Sep 15, 2015   ['tobyfox', '8-4']     4.2   
4     4        Hollow Knight  Feb 24, 2017      ['Team Cherry']     4.4   

  Times Listed Number of Reviews               Genres              Summary  \
0         3.9K              3.9K  ['Adventure', 'R...  Elden Ring is a ...   
1         2.9K              2.9K  ['Adventure', 'B...  A rogue-lite hac...   
2         4.3K              4.3K  ['Adventure', 'R...  The Legend of Ze...   
3         3.5K              3.5K  ['Adventure', 'I...  A small child fa...   
4           3K                3K  ['Adventure', 'I...  A 2D metroidvani...   

               Reviews Plays Playing Backlogs Wishlist  
0  ["The first play...   17K    3.8K     4.6K     4.8K  
1  ['convinced this...   21K    3.2K     6.3K     3.6K  
2  ['This game is t...   30K    2.5K       5K     2.6K  
3  ['soundtrack is ...   28K     679     4.9K     1.8K  
4  ["this games wor...   21K    2.4K     8.3K     2.3K